packages/commerce/backend/src/cache/in-memory.ts

```diff
@@ -0,0 +1,42 @@
+interface CacheData<TData> {
+  data: TData
+  expires: number | null
+  createdAt: number
+}
+
+export abstract class InMemoryCache<TData> {
+  protected cache: Map<string, CacheData<TData>>
+
+  constructor() {
+    this.cache = new Map()
+  }
+
+  set(key: string, value: TData, ttl?: number): void {
+    const now = Date.now()
+    const expires = ttl ? now + ttl : null
+    this.cache.set(key, { data: value, expires, createdAt: now })
+  }
+
+  async get(key: string): Promise<TData | null> {
+    const cached = this.cache.get(key)
+
+    if (!cached) {
+      return null
+    }
+
+    if (cached.expires !== null && cached.expires < Date.now()) {
```

Comment:
Since this is a base cache class and might be used in other places as well directly as an abstraction, do you intend to also cleanup the cache on different instances except for when the specific item is retrieved and found as expired?

How about we also have a loop that periodically checks expired items? (Unless we want to have a notion of observability on cache misses which i don't think is the case)

What do you think about either a loop that checks expired items, or an implementation of an LRU cache?

Reply:
I don't see how an LRU cache fits our case? Regarding the loop that periodically checks expired items, that's what I'm planning to do (we cache some order related fields so we won't need to save them in the database), but this is going to be in the second phase of the checkout flow (post-interaction).